mlflow/server/js/src/components/RequestStateWrapper.js

```diff
@@ -39,19 +43,21 @@ class RequestStateWrapper extends Component {
   }
 
   render() {
-    const { children } = this.props;
-    if (this.state.shouldRender) {
-      return <div>{children}</div>;
-    }
-    if (this.state.shouldRenderError) {
-      const errorRequests = RequestStateWrapper.getErrorRequests(this.props.requests);
-      const api = errorRequests.length > 0 ? errorRequests[0] : "";
-      console.log("ERROR", api.error);
-      return (
-        <div className="RequestStateWrapper-error">
-          {api.error.xhr.status}: {api.error.xhr.statusText}
-        </div>
-      );
+    const { children, errorRenderFunc } = this.props;
+    const { shouldRender, shouldRenderError } = this.state;
+    if (shouldRender) {
+      if (shouldRenderError) {
+        if (errorRenderFunc) {
+          const result = errorRenderFunc(this.props.requests);
+          if (result) {
+            return result;
+          }
+        }
+        // This triggers the OOPS error boundary.
+        throw Error("GOTO error boundary");
```

Comment:
I guess our new paradigm is to require components to define error handling in `errorRenderFunc` or fall through to here. It looks like before we did try to supply some extra info about the nature of the request failure in this default case e.g. via `api.error.xhr.status`, though the logic was broken (nice catch on that btw). Maybe we could still log something here? Although admittedly end users might not think to check the JS logs before raising an issue about a UI bug, so it might not help us too much to do so.

Reply:
Yeah logging something makes sense.